### DISCUSSION TITLE
How we can build a dedicated storage interface for FlexSearch?

### DISCUSSION THREAD

**Discussion by ts-thomas** [nextapps-de/flexsearch#418] [General]
Before some time I've started a Rust implementation of FlexSearch. Then I realized how good the Javascript version performs. It was not worth it to change over to Rust. But again I start thinking about to use Rust for a dedicated storage controller, perfectly optimized for FlexSearch Indexes. The goal is to extend the capacity of indexes (also data) by the filesystem and use the memory just for a well balanced cache or buffer.

I would like to know if anyone has experience with super fast storage implementations. Would be nice to do this in Rust or C++, but I'm looking forward for any other suggestion too.

**Comment by nickchomey** [comment]
I can't be of any assistance here, but can surrealdb provide any inspriation here? Particularly their WASM version for the browser. It is all written in Rust. https://github.com/surrealdb/surrealdb.wasm
